### DISCUSSION TITLE
How to properly use 3 way automatic merge?

### DISCUSSION THREAD

**Discussion by schierlm** [WinMerge/winmerge#1156] [Q&A] [answered]
Today I tried 3 way automatic merge, and it totally messed up my files. I opened the files in the order _base, _mine, _theirs and pressed merge in the _mine window. The conflicts became conflicts, but lines that were both changed in _mine and _theirs got replaced by the _base version, and changes that were only in _mine got lost.

By trial and error, I found out that when opening the files as _base, _theirs, _mine, or _mine, _theirs, _base, merging works. But I don't understand why and how it is supposed to work.

My test files:

### _Base
```
1
Unchanged
2
Changed MINE
3
Changed THEIRS
4
Changed CONFLICT
5
Changed BOTH
6
```

### _Theirs
```
1
Unchanged
2
Changed MINE
3
Changed THEIRS NEW
4
Changed CONFLICT NEWT
5
Changed BOTH NEW
6
```

### _Mine
```
1
Unchanged
2
Changed MINE NEW
3
Changed THEIRS
4
Changed CONFLICT NEWM
5
Changed BOTH NEW
6
```

**Comment by sdottaka** [comment, marked as answer]
WinMerge's auto-merge assumes the following order of files, depending on which pane is currently active(It’s a little inconsistent...):

- Middle pane: assumes that the files are in the order [theirs]-[base]-[mine], and merges the unconflicted changes in left (theirs) and right (mine) into middle (base).
- Right pane: assumes that the files are in the order [base]-[theirs]-[mine], and merges the non-conflicting changes in middle(theirs) into right(mine).
- Left pane: assumes that the files are in the order [mine]-[theirs]-[base], and merges non-conflicting changes in the middle (theirs) into the left (mine).

**Reply by schierlm** [reply]
So to sum it up: If I want to modify the base (merge in both parties), put it in the middle. If I want to modify my version, put it either left or right, and put base on opposite side. Thanks :)
